### ISSUE TITLE
Pressing Ctrl+R and Ctrl+T results in no undo after

### ISSUE THREAD

**Issue by GoogleCodeExporter** [cutelyaware/magiccube4d#113]
```
(What steps will reproduce the problem?)
1. Accidentally press Ctrl+r or Ctrl+t
2. Press Ctrl+z

(What is the expected output? What do you see instead?)
The cube should go back to its state before the shortcut was pressed. Instead, 
the cube remains solved and the file has to be reloaded/re-opened.

(What version of the product are you using? On what operating system?)
MC4D 4.0.185 on Windows 7

(Please provide any additional information below.)
An easier way to fix this is to put a Yes/No dialog box asking if the cube 
should be reset/solved by computer. (I know that it's easier to just save more 
often but I think that the dialog will make a more efficient solve)
```

Original issue reported on code.google.com by `Thermost...@gmail.com` on 10 Jul 2013 at 12:11

**Comment by SharkRetriever** [comment]
My alternate fix right now is to set the shortcuts to Ctrl+Alt+R and Ctrl+Alt+T to avoid accidentally triggering a reset/autosolve with one hand or two misaimed fingers.

However, being able to undo those would be cool for someone who wants to continuously show someone an autosolve over and over again, just saying.

**Comment by cutelyaware** [comment]
Resolved in v4.2. by changing the hotkeys. Reset is now <ctrl>0 which is fitting because it fits with the numbered scramble hotkeys <ctrl>1, <ctrl>2 etc. Zero random scrambles == reset. Cheat now uses <ctrl>L which was meant for the true solve. Maybe we'll get rid of the cheat version altogether when we have a good, short true solve. 

As for demos, I plan to add some "goto" controls including a "go to beginning" option (will stop at scramble boundary if any) and a "redo all" that will animate forward to the end as a better way to animate a solution.
